---
title: "Joining AA data to the complete_df data"
author: "Kaylah Marcello"
date: "2023-09-28"
output: 
  html_document: 
    keep_md: yes
---

# intall packages

## Load the Libraries

```{r message=FALSE}
library(tidyverse)
library(RColorBrewer)
library(paletteer)
library(janitor)
library(here)
library(skimr)
#library(viridis)
#library(shiny)
#library(shinydashboard)
#Ylibrary(devtools)
library(svglite)
#library(cluster)
#library(factoextra)
library(ggdendro)
library(grid)
library(gplots)
library(ggpmisc)
```

# NOTE: to highlight vertically 'Alt' + point and drag

```{r}
df <- read.csv("data/complete_0928.csv")
aa_df <- read.csv("data/TCS_0928.csv")
```

```{r}
head(df)
```
```{r}
head(aa_df)
```

```{r}
glimpse(df)
```

```{r}
glimpse(aa_df)
```

```{r}
clean_df <- clean_names(df)
clean_df
```

```{r}
clean_aa <- clean_names(aa_df)
clean_aa
```

```{r}
names(clean_df)
names(clean_aa)
```

```{r}
merged_df <- merge(clean_df, clean_aa, by = "seq", all = TRUE)
merged_df
```

```{r}
#df_CTG <- left_join(clean_df, clean_aa, keep = TRUE, by = 'ref_seq')
#df_CTG
```

```{r}
#names(df_CTG)
```

```{r}
#write_csv(df_CTG, "data/df_CTG.csv")
```

```{r}
write_csv(merged_df, "data/merged_df.csv")
```

## select columns of interest
```{r}
df_select <- merged_df %>% 
  select(organism_name,organism_tidy_name,seq,class,check_m_completeness,check_m_contamination,genome_size, gc_percentage,temp_cat,proline_count,lysine_count,r_k_ratio,aliphatic_percent_sum,aromaticity,flexibility_avg,flexibility_sum,gravy)
df_select
```

## proline count vs cyanobacteria boxplot
```{r}
df_proline <- df_select %>% 
  select(organism_name, proline_count, temp_cat) %>% 
  filter(!is.na(proline_count)) %>% 
  ggplot(aes(x=organism_name, y=proline_count, color=temp_cat))+
  geom_boxplot(alpha = .3, na.rm = TRUE, outlier.size = 0.2) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = rel(0.4)),
        panel.background = element_blank(),
        axis.title = element_text(size = 9),
        legend.title = element_text(size = 8),
        legend.text = element_text(size = 6))+
  scale_color_manual(values = c("warm-hot" = "lightsalmon", "cold-freezing" = "skyblue2", "temperate" = "darkseagreen")) +
  labs(title = "Proline Counts within TCS Genes of Cyanobacteria",
       x = "Cyanobacteria",
       y= "Proline Counts",
       color= "Temp Category")
  
df_proline

ggsave("TCS_proline.png")
```

## aromaticity vs cyanobacteria boxplot
```{r}
df_aromaticity <- df_select %>% 
  select(organism_name, aromaticity, temp_cat) %>% 
  filter(!is.na(aromaticity)) %>% 
  ggplot(aes(x=organism_name, y=aromaticity, color=temp_cat))+
  geom_boxplot(alpha = .3, na.rm = TRUE, outlier.size = 0.2) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = rel(0.4)),
        panel.background = element_blank(),
        axis.title = element_text(size = 9),
        legend.title = element_text(size = 8),
        legend.text = element_text(size = 6))+
  scale_color_manual(values = c("warm-hot" = "lightsalmon", "cold-freezing" = "skyblue2", "temperate" = "darkseagreen")) +
  labs(title = "Aromaticity of TCS Genes in Cyanobacteria",
       x = "Cyanobacteria",
       y= "Aromaticity",
       color= "Temp Category")

df_aromaticity

ggsave("TCS_aromaticity.png")
```

## argenine/lysine vs cyanobacteria boxplot
```{r}
df_r_k_ratio <- df_select %>% 
  select(organism_name, r_k_ratio, temp_cat) %>% 
  filter(!is.na(r_k_ratio)) %>% 
  ggplot(aes(x=organism_name, y=r_k_ratio, color=temp_cat))+
  geom_boxplot(alpha = .3, na.rm = TRUE, outlier.size = 0.2) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = rel(0.4)),
        panel.background = element_blank(),
        axis.title = element_text(size = 9),
        legend.title = element_text(size = 8),
        legend.text = element_text(size = 6))+
  scale_color_manual(values = c("warm-hot" = "lightsalmon", "cold-freezing" = "skyblue2", "temperate" = "darkseagreen")) +
  labs(title = "Argenine to Lysine ratio of TCS Genes in Cyanobacteria",
       x = "Cyanobacteria",
       y= "Aromaticity",
       color= "Temp Category")

df_aromaticity

ggsave("TCS_r_k_ratio.png")
```

## flexibility vs cyanobacteria boxplot
```{r}
df_flexibility <- df_select %>% 
  select(organism_name, flexibility_avg, temp_cat) %>% 
  filter(!is.na(flexibility_avg)) %>% 
  ggplot(aes(x=organism_name, y=flexibility_avg, color=temp_cat))+
  geom_boxplot(alpha = .3, na.rm = TRUE, outlier.size = 0.2) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = rel(0.4)),
        panel.background = element_blank(),
        axis.title = element_text(size = 9),
        legend.title = element_text(size = 8),
        legend.text = element_text(size = 6))+
  scale_color_manual(values = c("warm-hot" = "lightsalmon", "cold-freezing" = "skyblue2", "temperate" = "darkseagreen")) +
  labs(title = "Average Flexibility of Amino Acids Within TCS Genes of Cyanobacteria",
       x = "Cyanobacteria",
       y= "Proline Counts",
       color= "Temp Category")
  
df_proline

ggsave("TCS_flex_avg.png")
```

## gravy vs cyanobacteria boxplot with facet_wrap()
```{r}
df_gravy <- df_select %>% 
  select(organism_name, gravy, temp_cat) %>% 
  filter(!is.na(gravy)) %>% 
  ggplot(aes(x=organism_name, y=gravy, color=temp_cat))+
  facet_wrap(~temp_cat, ncol=4) +
  geom_boxplot(alpha = .3, na.rm = TRUE, outlier.size = 0.2) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = rel(0.25)),
        panel.background = element_blank(),
        axis.title = element_text(size = 9),
        legend.title = element_text(size = 8),
        legend.text = element_text(size = 6))+
  scale_color_manual(values = c("warm-hot" = "lightsalmon", "cold-freezing" = "skyblue2", "temperate" = "darkseagreen")) +
  labs(title = "GRAVY of TCS Genes in Cyanobacteria",
       x = "Cyanobacteria",
       y= "GRAVY",
       color= "Temp Category")
  
df_flexibility

ggsave("TCS_gravy.png")
```

## This one needs to be cleaned up
```{r}
#genome_size <- df_column %>% 
#  select(gravy, genome_size, temp_cat, genus, species) %>% 
#  filter(!is.na(gravy)) %>% 
#  ggplot(aes(x=species, y=gravy, color=temp_cat))+
#  facet_grid(genus~temp_cat) +
#  geom_point(size=0.2)+
#  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = rel(0.25)),
#        panel.background = element_blank(),
#        axis.title = element_text(size = 9),
#        legend.title = element_text(size = 8),
#        legend.text = element_text(size = 6))+
#  scale_color_manual(values = c("warm-hot" = "lightsalmon", "cold-freezing" = "skyblue2", #"temperate" = "darkseagreen")) +
#  labs(title = "GRAVY of TCS Genes in Cyanobacteria",
#       x = "Genome Size",
#       y= "GRAVY",
#       color= "Temp Category")
#
#genome_size
#  
#ggsave("genome_gravy.png")
```

## separating the organism_tidy_name into genus, species and strains.
```{r}
df_genus_species <- df_select %>% 
  separate(organism_tidy_name, into = c("genus", "species", "strain", "strain1", "strain2"), sep = "\\_")
df_genus_species
```

```{r}
df_TCSclass <- df_select %>% 
  separate(class, into = c("id", "type", "class"), sep = "\\|")
df_TCSclass
```


```{r}
df_gravy1 <- df_genus_species %>% 
  select(genus, gravy, species, temp_cat) %>% 
  filter(!is.na(gravy)) %>% 
  ggplot(aes(x=genus, y=gravy, color=temp_cat))+
  facet_grid(~temp_cat) +
  geom_boxplot(alpha = .3, na.rm = TRUE, outlier.size = 0.2) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, size = rel(0.25)),
        panel.background = element_blank(),
        axis.title = element_text(size = 9),
        legend.title = element_text(size = 8),
        legend.text = element_text(size = 6))+
  scale_color_manual(values = c("warm-hot" = "lightsalmon", "cold-freezing" = "skyblue2", "temperate" = "darkseagreen")) +
  labs(title = "GRAVY of TCS Genes in Cyanobacteria",
       x = "Cyanobacteria Genera",
       y= "GRAVY",
       color= "Temp Category")
  
df_gravy1

ggsave("TCS_gravy.png")
```

